server/bin/pbench-backup-tarballs.py

```diff
@@ -79,14 +80,16 @@ def sanity_check(lb_obj, s3_obj, config, logger):
         os.mkdir(backup)
     except FileExistsError:
         # directory already exists, verify it
-        backuppath = os.path.realpath(backup)
-        if not backuppath:
+        backup = Path(backup)
+        try:
+            backuppath = backup.resolve(strict=True)
+        except FileNotFoundError:
             logger.error(
                 "The BACKUP directory {}, does not resolve to a real location", backup
             )
             lb_obj = None
 
-        if not os.path.isdir(backup):
+        if not backup.is_dir():
```

Comment:
Shouldn't this be `backuppath.is_dir()`?

Reply:
I used backup as it was used earlier as well.

Reply:
I think the general rule is to check whether the fully resolved target is a directory.  So I think the previous checking of `backup` is a bug.